### PR TITLE
feat: simplify channel accounting, refactor state transitions

### PR DESCRIPTION
**Pull request by kincaidoneil**
#### Balances

- Balance logic is a no-op on client, must call `sendMoney`
- No balance config required on server: by default, requires prefunding & auto settles to 0
- Different client/server behavior temporarily complicates the balance logic until the connector refactor is complete
- Channels are not auto created on client and channel fees are not accounted in the balance (much simpler accounting, but not a permanent solution)
- BREAKING: balances for existing accounts are incompatible

#### Internal refactor

- Dual queues for incoming/outgoing channels
- All channel/claim _state_ updates are queued (previously a shared cached would be mutated, but these updates weren't queued and were hard to reason about)
- Single data structure for channels & claims
- State transitions much more straightforward (but will continue to iterate)
- BREAKING: new channel/claim schema
- BREAKING: claims are sent as BTP type MESSAGE instead of type TRANSFER

(Will add the auto-funding functionality to the server at a later commit)

### CODE REVIEW

File: src/account.ts [hunk 1]
```diff
@@ -1,645 +1,733 @@
...
+
+  /** Incoming amount owed to us by our peer for their packets we've forwarded */
+  receivableBalance: BigNumber
+
+  /** Outgoing amount owed by us to our peer for packets we've sent to them */
+  payableBalance: BigNumber
+
+  /**
+   * Amount of failed outgoing settlements that is owed to the peer, but not reflected
+   * in the payableBalance (e.g. due to sendMoney calls on client)
```
Comment (karzak) [inline src/account.ts:83]: I find this description confusing, don't know how to make it better though.

File: src/index.ts [hunk 2]
```diff
@@ -305,19 +321,27 @@ export default class EthereumPlugin extends EventEmitter2 implements PluginInsta
...
+  async sendData(data: Buffer) {
     return this._plugin.sendData(data)
   }
 
-  async sendMoney () {
-    this._log.error('sendMoney is not supported: use plugin balance configuration instead of connector balance for settlement')
+  async sendMoney(amount: string) {
+    const peerAccount = this._accounts.get('peer')
+    if (peerAccount) {
+      // If the plugin is acting as a client, enable sendMoney (required for prefunding)
```
Comment (karzak) [inline src/index.ts:351]: Should update the README, which currently says sendMoney is always a no-op.

**Review by karzak — APPROVED** [review]

File: src/account.ts [hunk 3]
```diff
@@ -1,645 +1,733 @@
...
+        )
 
-      const spentTooMuch = amountLeftoverWei.lt(0)
-      if (spentTooMuch) {
-        this.master._log.error(`Critical settlement error: spent ${spent}, above budget of ${budget} with ${account}`)
+      if (requiresTopUp && sufficientIncoming) {
+        return cachedChannel
+          ? this.depositToChannel(
+              cachedChannel,
+              this.master._outgoingChannelAmount.minus(cachedChannel.value)
```
Comment (karzak) [inline src/account.ts]:
```suggestion
              this.master._outgoingChannelAmount.minus(remainingInChannel(cachedChannel))
```

Comment (kincaidoneil) [reply src/account.ts]: Resolved by 39c1c48e31a33497d73c20909e6a0ff0ce7c6e3e

File: src/account.ts (hunk 1 above)
Comment (kincaidoneil) [reply src/account.ts:83]: Yeah :/ Hopefully it'll become simpler after the payable/receivable balances move to the connector.

**Pull request merged**